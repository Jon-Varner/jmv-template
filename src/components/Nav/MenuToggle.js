import React from 'react';
import PropTypes from 'prop-types';

const menuToggle = ({ menuOpen, toggleMenu }) => (
  <button
    className="menu-toggle"
    aria-label="Open the menu"
    onClick={toggleMenu}
  >
    <span className="menu-hamburger" aria-hidden="true" />
    <span className="menu-hamburger" aria-hidden="true" />
    <span className="menu-hamburger" aria-hidden="true" />
  </button>
);

menuToggle.propTypes = {
  menuOpen: PropTypes.bool,
  toggleMenu: PropTypes.func
};

export default menuToggle;
