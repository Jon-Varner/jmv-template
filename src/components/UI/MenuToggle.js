import React from 'react';
import PropTypes from 'prop-types';

const MenuToggle = ({ toggleMenu }) => (
  <button
    className="menu-toggle"
    aria-label="Open the menu"
    onClick={toggleMenu}
  >
    <span className="menu-hamburger" aria-hidden="true" />
    <span className="menu-hamburger" aria-hidden="true" />
    <span className="menu-hamburger" aria-hidden="true" />
  </button>
);

MenuToggle.propTypes = {
  toggleMenu: PropTypes.func
};

export default MenuToggle;
