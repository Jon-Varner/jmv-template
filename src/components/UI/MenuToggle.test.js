import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import MenuToggle from './MenuToggle';

afterEach(cleanup);

const toggleMenu = jest.fn();

it('renders', () => {
  const { asFragment } = render(<MenuToggle toggleMenu={toggleMenu} />);
  expect(asFragment()).toMatchSnapshot();
});
