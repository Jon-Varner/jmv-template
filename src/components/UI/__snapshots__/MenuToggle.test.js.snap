// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    aria-label="Open the menu"
    class="menu-toggle"
  >
    <span
      aria-hidden="true"
      class="menu-hamburger"
    />
    <span
      aria-hidden="true"
      class="menu-hamburger"
    />
    <span
      aria-hidden="true"
      class="menu-hamburger"
    />
  </button>
</DocumentFragment>
`;
